### PR TITLE
[gn] changes for fuchsia

### PR DESCRIPTION
**Pull request by paragdixit-g**
This change fixes some build issues faced in Fuchsia when trying to update OpenThread repo to latest changes. 

It fixes one issue in mbedtls where it is missing a file in BUILD.gn
Adds BUILD.gn target for src/ncp 
Fixes a warning/error of unused variable.

### CODE REVIEW

File: src/ncp/BUILD.gn [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+import("../../etc/gn/openthread.gni")
+
+source_set("openthread-ncp-sources") {
```
Comment (mspang) [inline src/ncp/BUILD.gn:30]: This should be a static_library called libopenthread-{rcp,ncp-ftd,ncp-mtd} (depending on configuration).

File: src/ncp/BUILD.gn [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    "ncp_base_radio.cpp",
+    "ncp_spi.cpp",
+    "ncp_spi.hpp",
+    "ncp_uart.cpp",
+    "ncp_uart.hpp",
+  ]
+
+  public_configs = [ ":ncp_config" ]
+
+  public_deps = [ "../core:libopenthread_platform" ]
```
Comment (mspang) [inline src/ncp/BUILD.gn:49]: Deps should be libopenthread-{ftd, mtd, radio}

**Review by mspang — COMMENTED** [review]
This isn't consistent with the other build systems or with the other GN build files.

File: BUILD.gn [hunk 3]
```diff
@@ -102,6 +102,10 @@ group("libopenthread-spinel-rcp") {
 
 if (current_os == "fuchsia") {
   group("lib-ot-core") {
-    public_deps = [ ":libopenthread-ftd" ]
+    public_deps = [
```
Comment (mspang) [inline BUILD.gn:105]: nit: Run "gn format" on this file (and the others).

**Review by mspang — APPROVED** [review]

File: BUILD.gn (hunk 3 above)
Comment (paragdixit-g) [reply BUILD.gn:105]: Thank you Michael, 
I ran "gn format" on all the files.

File: src/ncp/BUILD.gn (hunk 2 above)
Comment (paragdixit-g) [reply src/ncp/BUILD.gn]: Done

File: src/ncp/BUILD.gn (hunk 1 above)
Comment (paragdixit-g) [reply src/ncp/BUILD.gn]: Done

**Review by jwhui — APPROVED** [review]

File: src/lib/spinel/radio_spinel_impl.hpp [hunk 4]
```diff
@@ -360,6 +360,10 @@ otError RadioSpinel<InterfaceType, ProcessContextType>::CheckRadioCapabilities(v
     {
         otRadioCaps missingCaps = (mRadioCaps & kRequiredRadioCaps) ^ kRequiredRadioCaps;
 
+        // missingCaps may be an unused variable when otLogCritPlat is blank
+        // avoid compiler warning in that case
+        (void)missingCaps;
```
Comment (jwhui) [inline src/lib/spinel/radio_spinel_impl.hpp:365]:
```suggestion
        OT_UNUSED_VARIABLE(missingCaps);
```

**Pull request merged**